test/test_rosdep_pip.py

```diff
@@ -67,6 +67,7 @@ def test_PipInstaller_get_depends():
     assert ['foo'] == installer.get_depends(dict(depends=['foo']))
 
 
+@patch.dict(os.environ, {'PIP_BREAK_SYSTEM_PACKAGES': '1'})
```

Comment:
would be nice to test that the command fails if this isn't set and the version is >= 3.11, but you'd also need to make sure that the other config methods are not used, or just patch `externally_managed_installable()`, but I would leave that up to you.

Reply:
I'm limited somewhat by my lack of experience with Python's mocking. Let me see if I can skim enough examples and read enough of the docs to stub the method for a few tests before.

It's a good push. Tests added in 8b48f13